### PR TITLE
Auto-apply milestones

### PR DESCRIPTION
**Pull request by Maxr1998**
Tested in Maxr1998/jellyfin-android#1.

### CODE REVIEW

File: .github/workflows/repo-milestone.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - reopened
+
+jobs:
+  milestone:
+    name: Apply milestone
+    runs-on: ubuntu-22.04
+    if: ${{ contains(github.repository_owner, 'jellyfin') }}
+    steps:
+      - uses: actions/github-script@v6
+        id: my-script
```
Comment (nielsvanvelzen) [inline .github/workflows/repo-milestone.yaml:16]: I believe the id can be removed if it's not used in the workflow

**Review by nielsvanvelzen — COMMENTED** [review]
I think this script will fail if we make a milestone for the next minor release (0.15.0) and then decide to create another patch release (0.14.5) because the latter will be the newest.

**Comment by Maxr1998** [comment]
It'll use the oldest open/active milestone (0.15.0) which, in my opinion, is the right choice. If there are PRs directly going to the patch release, you can still manually override the milestone.

File: .github/workflows/repo-milestone.yaml (hunk 1 above)
Comment (Maxr1998) [reply .github/workflows/repo-milestone.yaml]: Fixed.

File: .github/workflows/repo-milestone.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            );
+
+            if (milestone === null) {
+              console.warn('No suitable milestone found, aborting.');
+              return;
+            }
+
+            console.log('Found milestone to apply: ' + milestone.title);
+
+            github.rest.issues.update({
```
Comment (nielsvanvelzen) [inline .github/workflows/repo-milestone.yaml:42]: This is missing an await

File: .github/workflows/repo-milestone.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+              (prev, current) => prev?.number < current.number ? prev : current, 
+              null,
+            );
+
+            if (milestone === null) {
+              console.warn('No suitable milestone found, aborting.');
+              return;
+            }
+
+            console.log('Found milestone to apply: ' + milestone.title);
```
Comment (nielsvanvelzen) [inline .github/workflows/repo-milestone.yaml:40]:
```suggestion
            console.log(`Found milestone to apply: ${milestone.title}`);
```

File: .github/workflows/repo-milestone.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+            github.rest.issues.update({
+              owner,
+              repo,
+              issue_number: pr_id,
+              milestone: milestone.number,
+            });
+            
+            console.log('Successfully applied milestone ' + milestone.title + ' to PR #' + pr_id);
```
Comment (nielsvanvelzen) [inline .github/workflows/repo-milestone.yaml:49]:
```suggestion
            console.log(`Successfully applied milestone ${milestone.title} to PR #${pr_id}`);
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**